interfaces/acados_template/acados_template/c_templates_tera/acados_mex_create.in.c

```diff
@@ -71,37 +70,30 @@ void mexFunction(int nlhs, mxArray *plhs[], int nrhs, const mxArray *prhs[])
     ocp_nlp_solver *nlp_solver = acados_get_nlp_solver();
     void *nlp_opts = acados_get_nlp_opts();
 
-    mexPrintf("acados: got pointer to objectes!\n");
-    
-
+    // mexPrintf("acados: got pointer to objectes!\n");
 
     // field names of output struct
-    char *fieldnames[7];
-    fieldnames[0] = (char*)mxMalloc(50);
-    fieldnames[1] = (char*)mxMalloc(50);
-    fieldnames[2] = (char*)mxMalloc(50);
-    fieldnames[3] = (char*)mxMalloc(50);
-    fieldnames[4] = (char*)mxMalloc(50);
-    fieldnames[5] = (char*)mxMalloc(50);
-    fieldnames[6] = (char*)mxMalloc(50);
-
-    memcpy(fieldnames[0],"plan",sizeof("plan"));
-    memcpy(fieldnames[1],"config",sizeof("config"));
-    memcpy(fieldnames[2],"dims",sizeof("dims"));
-    memcpy(fieldnames[3],"opts",sizeof("opts"));
-    memcpy(fieldnames[4],"in",sizeof("in"));
-    memcpy(fieldnames[5],"out",sizeof("out"));
-    memcpy(fieldnames[6],"solver",sizeof("solver"));
-    // create output struct
-    plhs[0] = mxCreateStructMatrix(1, 1, 7, (const char **) fieldnames);
-
-    mxFree( fieldnames[0] );
-    mxFree( fieldnames[1] );
-    mxFree( fieldnames[2] );
-    mxFree( fieldnames[3] );
-    mxFree( fieldnames[4] );
-    mxFree( fieldnames[5] );
-    mxFree( fieldnames[6] );
+    int fields_ocp = 8;
+    int fields_ext_fun = 11;
+    int nfields = MAX(fields_ext_fun, fields_ocp);
+    char *fieldnames[nfields];
```

Comment:
This is a variable size array, which we avoided to use everywhere in acados (core), since this gives issues with e.g. visual studio compiler. Not sure if it is safe for portability to use it here...